cpp/docs/DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,823 @@
+# libcudf C++ Developer Guide
+
+This document serves as a guide for contributors to libcudf C++ code. Developers should also refer 
+to these additional files for further documentation of `libcudf` best practices.
+
+* [Documentation Guide](DOCUMENTATION.md) for guidelines on documenting libcudf code.
+* [Testing Guide(TESTING.md) for guidelines on writing unit tests.
+* [Benchmarking Guide](TODO) for guidelines on writing unit benchmarks.
+
+# Overview
+
+libcudf is a C++ library that provides GPU-accelerated data-parallel algorithms for processing 
+column-oriented tabular data. Algorithms provided range from slicing and filtering to sorting, 
+and various types of aggregations, to database type operations such as grouping and joins. libcudf
+serves a number of clients via multiple language interfaces, including Python and Java. Users may
```

Comment:
The structure of this sentence is a bit odd "From x & y TO z & A TO types of B & C"
```suggestion
column-oriented tabular data. Algorithms provided range from slicing, filtering, sorting 
and various types of aggregations, to database type operations such as grouping and joins. libcudf
serves a number of clients via multiple language interfaces, including Python and Java. Users may
```